probe-rs/src/debug/debug_info.rs

```diff
@@ -1120,6 +1121,35 @@ impl DebugInfo {
     }
 }
 
+/// This is a fuzzy 'equality' check for path names.
+/// If both paths are absolute, then it will use the standard `Path::eq` method.
+/// If the secondary path is relative, then it will succeeed if the primary ends with the compared path,
+/// e.g. `/home/user/project/src/main.rs` ends with `src/main.rs` and will return true.
+pub(crate) fn path_fuzzy_eq(primary_path: &Path, secondary_path: &Path) -> bool {
+    if primary_path.is_absolute() && secondary_path.is_relative() {
+        // First make sure the secondary path does not start with special sequences, e.g. `./` or `../`
```

Comment:
What happens if a `..` occurs in the middle of a path? Why is a `..`at the beginning handled in a special way?

Reply:
Could be fixed with `dunce` I guess :)

Reply:
@Tiwalun, I accounted for the ones in the front, because it was relatively simple. For more complex cases, it won't find a match. Future enhancement :)

@Yatekii, I looked at `dunce`, and I'm not sure this will help. For one thing, it says it only acts on Windows paths. 

Any other suggestions?

Reply:
For non-windows https://doc.rust-lang.org/std/path/struct.Path.html#method.canonicalize works just fine :)

Reply:
Oooh ... nice find.  I will implement and test.  Thanks.

Reply:
@Yatekii (cc: @Tiwalun ) thanks for the suggestion. The `canonicalize` does a really nice job of producing cleansed absolute paths and resolves the various forms of relative paths in all my test cases.